src/utils/CameraUtils.js

```diff
@@ -0,0 +1,448 @@
+    let resolve;
+    return { promise: new Promise((r) => { resolve = r; }), resolve };
+};
+
+// Wrap angle in degrees to [-180 180]
+function wrapTo180(angle) {
+    return angle - Math.floor((angle + 179.0) / 360) * 360;
+}
+
+function tileLayer(view) {
+    return view.getLayers(l => l.protocol == 'tile')[0];
+}
+
+// fix coord.altitude < 0 and !pickedPosition (with plane and ellipsoid intersection)
+function getGroundTargetFromCamera(view, camera, target) {
+    camera.updateMatrixWorld(true);
+    // FIXME view.wgs84TileLayer
+    const pickedPosition = view.wgs84TileLayer ? view.getPickingPositionFromDepth() : undefined;
+    const range = pickedPosition && !isNaN(pickedPosition.x) ? camera.position.distanceTo(pickedPosition) : 100;
+    camera.localToWorld(target.set(0, 0, -range));
+}
+
+function proxyProperty(view, camera, rig, key) {
+    rig.proxy.position[key] = camera.position[key];
+    Object.defineProperty(camera.position, key, {
+        get: () => rig.proxy.position[key],
+        set: (newValue) => {
+            rig.removeProxy(view, camera);
+            camera.position[key] = newValue;
+        },
+    });
+}
+
+// the rig is used to manipulate the camera
+// It consists of a tree of 3D objects, each element is assigned a task
+//
+//                      Transformation
+//
+// rig                  position on Coordinate  (for the globe is rotation)
+// |
+// └── sealevel         position on altitude zero
+//     |
+//     └── target       position on DEM, and rotation (pitch and heading)
+//         |
+//         └── camera   distance to target
+//
+// When all transformations are calculated,
+// this.camera's transformation is applied to view.camera.camera
+class CameraRig extends THREE.Object3D {
+    constructor() {
+        super();
+        // seaLevel is on rig's z axis, it's at altitude zero
+        this.seaLevel = new THREE.Object3D();
+        // target is on seaLevel's z axis and target.position.z is the DEM altitude
+        this.target = new THREE.Object3D();
+        this.target.rotation.order = 'ZXY';
+        // camera look at target
+        this.camera = new THREE.Camera();
+        this.add(this.seaLevel);
+        this.seaLevel.add(this.target);
+        this.target.add(this.camera);
+        // sea level's geograohic coordinate
+        this.coord = new Coordinates('EPSG:4978', 0, 0);
+        // sea level's worldPoistion
+        this.targetWorldPosition = new THREE.Vector3();
+    }
+
+    // apply rig.camera's transformation to camera
+    applyTransformToCamera(view, camera) {
+        if (this.proxy) {
+            camera.quaternion.onChange(() => {});
+            this.camera.matrixWorld.decompose(this.proxy.position, camera.quaternion, camera.scale);
+            camera.quaternion.onChange(() => this.removeProxy(view, camera));
+        } else {
+            this.camera.matrixWorld.decompose(camera.position, camera.quaternion, camera.scale);
+        }
+    }
+
+    setProxy(view, camera) {
```

Comment:
Could you elaborate more on the role of `proxy` ?

Reply:
see  in cameraTransformOptions's documentation